src/System.Globalization/tests/CharUnicodeInfo/CharUnicodeInfoTestData.cs

```diff
@@ -0,0 +1,224 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace System.Globalization.Tests
+{
+    public static class CharUnicodeInfoTestData
+    {
+        private static List<CharUnicodeInfoTestCase> s_testCases;
+        public static List<CharUnicodeInfoTestCase> TestCases
+        {
+            get
+            {
+                if (s_testCases == null)
+                {
+                    s_testCases = new List<CharUnicodeInfoTestCase>(); 
+                    string fileName = "UnicodeData.8.0.txt";
+                    Stream stream = typeof(CharUnicodeInfoGetUnicodeCategoryTests).GetTypeInfo().Assembly.GetManifestResourceStream(fileName);
+                    using (StreamReader reader = new StreamReader(stream))
+                    {
+                        while (!reader.EndOfStream)
+                        {
+                            s_testCases.Add(Parse(reader.ReadLine()));
+                        }
+                    }
+                }
+
+                return s_testCases;
```

Comment:
This isn't thread-safe, and if two tests in different classes try to use this concurrently, it's likely one will fail.  You can use ```Lazy<T>``` to handle it, e.g.
```C#
private static readonly Lazy<List<CharUnicodeInfoTestCase>> s_testCases = new Lazy<CharUnicodeInfoTestCase>(() =>
{
    var testCases = new List<CharUnicodeInfoTestCase>();
    string fileName = ..
    ...
    return testCases;
});
public static List<CharUnicodeInfoTestCase> TestCases => s_testCases.Value;
```